silkrpc/commands/eth_api.cpp

```diff
@@ -194,6 +194,51 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_block_by_number(const nlohm
     co_return;
 }
 
+// https://eth.wiki/json-rpc/API#eth_getBlockTransactionCountByNumber
+asio::awaitable<void> EthereumRpcApi::handle_eth_get_block_transaction_count_by_number(const nlohmann::json& request, nlohmann::json& reply) {
+    auto params = request["params"];
+    if (params.size() != 1) {
+        auto error_msg = "invalid eth_getBlockTransactionCountByNumber params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+    const auto block_number_or_name = params[0].get<std::string>();
+    SILKRPC_DEBUG << "block_number_or_name: " << block_number_or_name << "\n";
+
+    auto tx = co_await database_->begin();
+
+    try {
+        ethdb::kv::TransactionDatabase tx_database{*tx};
+
+        uint64_t block_number_extended;
+        if (block_number_or_name == "earliest") {
+            block_number_extended = core::kEarliestBlockNumber;
+        } else if (block_number_or_name == "latest") {
+            block_number_extended = core::kLatestBlockNumber;
+        } else if (block_number_or_name == "pending") {
+            block_number_extended = core::kPendingBlockNumber;
+        } else {
+            block_number_extended = std::stol(block_number_or_name, 0, 16);
+        }
+        const auto block_number = co_await core::get_block_number(block_number_extended, tx_database);
+        const auto block_with_hash = co_await core::rawdb::read_block_by_number(tx_database, block_number);
+        const auto total_difficulty = co_await core::rawdb::read_total_difficulty(tx_database, block_with_hash.hash, block_number);
+        const Block extended_block{block_with_hash, total_difficulty, false};
```

Comment:
Lines 226 and 227 are unuseful because `block_with_hash` already contains the transactions; moreover, line 226 performs a database lookup that is unnecessary for this method implementation. So both lines must be removed